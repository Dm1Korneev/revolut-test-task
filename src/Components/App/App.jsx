import React from 'react';

import { Container } from './App.styled';

function App() {
  return (
    <Container />
  );
}

export default App;
