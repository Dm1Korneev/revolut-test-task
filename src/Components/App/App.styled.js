import styled from 'styled-components';

export const Container = styled.main`
    margin: 0 auto;
    width: 500px;
    height: 500px;
    background-color: red;
`;
