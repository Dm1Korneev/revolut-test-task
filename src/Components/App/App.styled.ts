import styled from 'styled-components';

export const Container = styled.main`
    margin: 0 auto;
    width: fit-content;
    padding-top: 2rem;
`;
