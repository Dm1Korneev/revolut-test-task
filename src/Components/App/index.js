import commonHoc from 'Components/commonHoc';

import App from './App';

const mapStateToProps = () => ({
});

const mapDispatchToProps = {
};

export default commonHoc(App, {
  mapStateToProps,
  mapDispatchToProps,
});
