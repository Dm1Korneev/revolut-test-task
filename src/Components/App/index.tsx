import React, { FC } from 'react';

import CurrencyConverter from 'Components/CurrencyConverter';
import { Container } from './App.styled';

const App: FC = (): JSX.Element => (
  <Container>
    <CurrencyConverter />
  </Container>
);

export default App;
