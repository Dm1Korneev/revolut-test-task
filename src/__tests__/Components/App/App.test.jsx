import React from 'react';
import { render } from '@testing-library/react';
import App from 'Components/App/';

jest.mock('Components/CurrencyConverter', () => global.mockComponent('CurrencyConverter'));

describe('App component', () => {
  test('renders the component', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
