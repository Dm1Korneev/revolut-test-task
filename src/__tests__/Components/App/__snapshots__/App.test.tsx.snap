// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa kWexiN"
  >
    <div
      data-originalcomponent="CurrencyConverter"
    />
  </main>
</DocumentFragment>
`;
